### PR TITLE
[docs] Cleanup the pickers migration guide

### PR DESCRIPTION
**Pull request by flaviendelangle**

### CODE REVIEW

File: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md [hunk 1]
```diff
@@ -141,19 +141,14 @@ these props will now be passed to the hidden `<input />` element.
 If you are passing a custom field component to your pickers, you need to create a new one that is using the accessible DOM structure.
 This new component will need to use the `PickersSectionList` component instead of an `<input />` HTML element.
 
-You can have a look at the [Using a custom input](/x/react-date-pickers/custom-field/#using-a-custom-input) to have a concrete example.
-
-:::info
```
Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v7/migration-pickers-v7.md:146]: This is not going to happen :laughing:

Comment (LukasTy) [reply docs/data/migration/migration-pickers-v7/migration-pickers-v7.md:146]: Clearly. 😆

File: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md [hunk 2]
```diff
@@ -293,7 +288,28 @@ The following variables were renamed to have a coherent `Picker` / `Pickers` pre
   + const pickerContext = usePickerContext();
   ```
 
-## Removed types
+## Typing breaking changes
+
+### Remove `TDate` generic
+
+The `TDate` generic has been removed of all the types, interfaces and variables of the `@mui/x-date-pickers` and `@mui/x-date-pickers-pro` packages.
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v7/migration-pickers-v7.md:295]:
```suggestion
The `TDate` generic has been removed from all the types, interfaces, and variables of the `@mui/x-date-pickers` and `@mui/x-date-pickers-pro` packages.
```

File: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md [hunk 3]
```diff
@@ -293,7 +288,28 @@ The following variables were renamed to have a coherent `Picker` / `Pickers` pre
...
+
+If you were passing your date object as a generic to any element of one of those packages, you can remove it:
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v7/migration-pickers-v7.md:297]:
```suggestion
If you were passing your date object type as a generic to any element of one of those packages, you can remove it:
```

File: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md [hunk 4]
```diff
@@ -293,7 +288,28 @@ The following variables were renamed to have a coherent `Picker` / `Pickers` pre
...
++<DatePicker value={value} onChange={onChange} />
+
+-type Slot = DateCalendarSlots<Dayjs>['calendarHeader'];
++type Slot = DateCalendarSlots['calendarHeader'];
+
+-type Props = DatePickerToolbarProps<Dayjs>;
++type Props = DatePickerToolbarProps;
+```
+
+The full list of the impacted element will be added to the migration guide in a follow up release.
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v7/migration-pickers-v7.md:310]:
```suggestion
A follow-up release will add the full list of the impacted elements to the migration guide.
```

**Review by LukasTy — APPROVED** [review]
LGTM. 👍

**Pull request merged**